src/main/php/CommandLine/ToolAdapters/PHPMessDetectorAdapter.php

```diff
@@ -1,85 +1,56 @@
 <?php
+
 namespace Zooroyal\CodingStandard\CommandLine\ToolAdapters;
 
 use Symfony\Component\Console\Output\OutputInterface;
 use Zooroyal\CodingStandard\CommandLine\Library\Environment;
 use Zooroyal\CodingStandard\CommandLine\Library\GenericCommandRunner;
 
-class PHPMessDetectorAdapter implements ToolAdapterInterface
+class PHPMessDetectorAdapter extends AbstractBlackAndWhitelistAdapter implements ToolAdapterInterface
 {
     /** @var string */
-    private $messDetectCommandWhitelist;
+    protected $blacklistToken = '.dontMessDetectPHP';
     /** @var string */
-    private $messDetectCommandBlacklist;
-    /** @var OutputInterface */
-    private $output;
-    /** @var Environment */
-    private $environment;
-    /** @var GenericCommandRunner */
-    private $genericCommandRunner;
+    protected $filter = '.php';
     /** @var string */
-    private $stopword;
+    protected $blacklistGlue = ',';
     /** @var string */
-    private $filter;
+    protected $whitelistGlue = ',';
+
 
     /**
      * PHPCodeSnifferAdapter constructor.
      *
-     * @param Environment          $environment
-     * @param OutputInterface      $output
+     * @param Environment $environment
+     * @param OutputInterface $output
      * @param GenericCommandRunner $genericCommandRunner
      */
     public function __construct(
         Environment $environment,
         OutputInterface $output,
         GenericCommandRunner $genericCommandRunner
     ) {
-        $this->environment          = $environment;
-        $this->output               = $output;
+        $this->environment = $environment;
+        $this->output = $output;
         $this->genericCommandRunner = $genericCommandRunner;
 
         $phpMessDetectorConfig = $environment->getPackageDirectory() . '/src/config/phpmd/ZooRoyalDefault/phpmd.xml';
-        $rootDirectory         = $environment->getRootDirectory();
+        $rootDirectory = $environment->getRootDirectory();
 
-        $this->stopword = '.dontMessDetectPHP';
-        $this->filter   = '.php';
-
-        $this->messDetectCommandWhitelist = 'php ' . $rootDirectory . '/vendor/bin/phpmd %1$s' .
+        $this->commands['PHPMDWL'] = 'php ' . $rootDirectory . '/vendor/bin/phpmd %1$s' .
             ' text ' . $phpMessDetectorConfig . ' --suffixes php';
-        $this->messDetectCommandBlacklist = 'php ' . $rootDirectory . '/vendor/bin/phpmd '
+        $this->commands['PHPMDBL'] = 'php ' . $rootDirectory . '/vendor/bin/phpmd '
             . $rootDirectory . ' text ' . $phpMessDetectorConfig . ' --suffixes php --exclude %1$s';
     }
 
-    /**
-     * Search for violations by using PHPCS and write finds to screen.
-     *
-     * @param string $targetBranch
-     * @param bool   $processIsolation
-     *
-     * @return int|null
-     */
     public function writeViolationsToOutput($targetBranch = '', $processIsolation = false)
```

Comment:
Why do you remove the comment?

```suggestion
    public function writeViolationsToOutput(string $targetBranch = '', $processIsolation = false) : ?int
```

Reply:
Brain fart ^^